src/FuncSharp/Strings/NonEmptyString.cs

```diff
@@ -36,21 +39,79 @@ public static IOption<NonEmptyString> Create(string value)
         return Option.Valued(new NonEmptyString(value));
     }
 
+    #region Proxy methods to string
```

Comment:
is this really necessary to have here? 🤷‍

Reply:
Well either this or create it as an extension. It feels nicer here to me.

Reply:
like I said, IMO they are not neede as you expose the `Value` property anyway.

Reply:
But it creates a lot of ugly code when it's used in collections, dictionaries, Options, Tries. 
You have to have extra selects, maps, sometimes just create another dictionary from the dictionary you have, because you get a dictionary of `NonEmptyString` and you need a dictionary of `String` or the other way around. 

What's the negative side of having these when they make the users's life easier? It's not wrong, why not just have it.